### PR TITLE
Add Project and Package classes for storing recipe data, dependencies

### PR DESCRIPTION
**Pull request by shelbyspees**

### CODE REVIEW

**Comment by naftulikay** [comment]
@shelbyspees Here's an example output from `fpm-cook inspect` of a given package:

```
{
  "arch": null,
  "build_depends": [],
  "builddir": "/vagrant/recipes/project-name/tmp-build",
  "cachedir": "/vagrant/recipes/project-name/cache",
  "chain_package": null,
  "chain_recipes": [],
  "config_files": [],
  "conflicts": [],
  "datadir": "/vagrant/recipes/project-name/config",
  "default_prefix": null,
  "depends": [],
  "description": "Atlas Environment Generation Service",
  "destdir": "/vagrant/recipes/project-name/tmp-dest",
  "directories": [],
  "environment": {},
  "exclude": [],
  "fpm_attributes": {},
  "homepage": null,
  "license": null,
  "maintainer": "...",
  "md5": null,
  "name": "project-name",
  "omnibus_additional_paths": [],
  "omnibus_dir": null,
  "omnibus_package": null,
  "omnibus_recipes": [],
  "patches": [],
  "pkgdir": "/vagrant/recipes/project-name/pkg",
  "post_install": "scripts/post-install",
  "post_uninstall": null,
  "pre_install": null,
  "pre_uninstall": null,
  "provides": [],
  "replaces": [],
  "revision": "1",
  "rpm_dist": null,
  "section": "extra",
  "sha1": null,
  "sha256": null,
  "source": "/vagrant/recipes/project-name/README.md",
  "spec": {
    "with": "local_path"
  },
  "tmp_root": "/vagrant/recipes/project-name",
  "vendor": null,
  "version": "1.0.0",
  "workdir": "/vagrant/recipes/project-name"
}
```

You'll need to verify that your tool can parse this JSON output from each project to assemble your data objects. Most of this information is useful to us.

File: lib/revolution/project.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+require 'yaml'
+require 'fpm/cookery/recipe'
+
+class Project
+  attr_accessor :name
+
+  def self.initialize
+    @packages = []
+    puts '[Project initialized]'
+    Project.new
```
Comment (naftulikay) [inline lib/revolution/project.rb]: In `self.initialize`, you're going to want to call `fpm-cook inspect` to get JSON information as I have provided in the PR. This JSON information will tell about dependencies, package name, version, revision, etc.

File: lib/revolution/project.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    # @recipe = FPM::Cookery::Recipe.load(recipe_file)
+    @recipe = nil
+    # find_dependencies # FIXME: get dependencies
+    Package.new
+  end
+
+  def self.find_dependencies
+    puts '[Looking for dependencies]'
+    @dependencies = []
+    puts @recipe.depends # FIXME: not sure if this will work
```
Comment (naftulikay) [inline lib/revolution/project.rb]: Please try this in the wild.

**Comment by naftulikay** [comment]
@shelbyspees So at the very beginning of Revolution's run, it should take a parameter called `--recipe-root` which will point to the root directory of where packages are located, like so:

```
revolution --recipe-root recipes/
```

Revolution will then list that `recipes/` directory for all subdirectories, and then for each subdirectory, if it contains a `recipe.rb` file, it is indeed a project. 

Please check in with the previous code from packages where the previous engineer used `fpm-cook inspect` to determine facts about the build.

**Comment by shelbyspees** [comment]
Things to do:
- [X] Ingest JSON data
- [ ] Clean up JSON data ingestion to skip file IO step
- [ ] Test `recipe.depends`
- [ ] Accept `--recipe-root` parameter
- [ ] Manage `chain_recipes` in Project

File: examples/package-with-deps/recipe.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env ruby
+
+class PackageWithDeps < FPM::Cookery::Recipe
+
+  name        'package-with-deps'
+  description 'Project containing multiple packages for testing'
+  homepage    'http://intomore.com/'
```
Comment (smaslennikov) [inline examples/package-with-deps/recipe.rb]: Should be https://medium.com/@GrindrLabs

File: examples/project-chain-recipes/recipe.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env ruby
+
+class ProjectChainRecipes < FPM::Cookery::Recipe
+
+  name        'project-chain-recipes'
+  description 'Project containing multiple packages for testing'
+  homepage    'http://intomore.com/'
```
Comment (smaslennikov) [inline examples/project-chain-recipes/recipe.rb]: Should be https://medium.com/@GrindrLabs

**Pull request merged**